jupyterhub/tests/test_auth.py

```diff
@@ -103,6 +103,30 @@ def getgrnam(name):
     assert authorized is None
 
 
+@pytest.mark.gen_test
+def test_pam_auth_blacklist():
+    authenticator = MockPAMAuthenticator()
+    authorized = yield authenticator.get_authenticated_user(None, {
+        'username': 'wash',
+        'password': 'wash',
+    })
+    assert authorized['name'] == 'wash'
+
+    authenticator = MockPAMAuthenticator(blacklist={'wash'})
+    authorized = yield authenticator.get_authenticated_user(None, {
+        'username': 'wash',
+        'password': 'wash',
+    })
+    assert authorized is  None
+
+    authenticator = MockPAMAuthenticator(blacklist={'wash'}, whitelist={'wash', 'kaylee'})
```

Comment:
Maybe remove 'wash' from whitelist. I don't want a user being in both whitelist and blacklist to be supported.

Reply:
My thought was shouldn't be supported, but it must either default-deny (tested for) or raise an error (not tested)...

Reply:
OK

Reply:
We might consider (doesn't need to be in this PR) testing several combinations for expected behavior: user not in either list; user in one or the other; user in both